### ISSUE TITLE
Selectpicker is not working in asp

### ISSUE THREAD

**Issue by omerzia** [silviomoreto/bootstrap-select#720]
Hi all,

asp list item are not being viewed on aspx page.

  `<asp:DropDownList ID="DropDownList2" CssClass="selectpicker" data-style="btn-danger" runat="server"> `
     `<asp:ListItem>Action</asp:ListItem> `
   `<asp:ListItem>Another action</asp:ListItem> `
     `<asp:ListItem>Something else here</asp:ListItem> `
     `</asp:DropDownList> `
```javascript
<script type="text/javascript">
       $(document).ready(function () {
           $('.selectpicker').selectpicker('');
       });
    </script>
```
Any ideas?

**Comment by seanbotha123** [comment]
Been using bootstrap select with asp.net for a while now, code to make it work, while also adding 'option group' functionality to asp dropdownlist: ( VB code)
```
Public Class DropDownListAdapter
    Inherits System.Web.UI.WebControls.Adapters.WebControlAdapter
    Protected Overrides Sub RenderContents(writer As HtmlTextWriter)
       
            Dim list As DropDownList = TryCast(Me.Control, DropDownList)

            Dim currentOptionGroup As String
            Dim renderedOptionGroups As New List(Of String)()

            For Each item As ListItem In list.Items
                If item.Attributes("OptionGroup") Is Nothing Then
                    RenderListItem(item, writer)
                Else
                    currentOptionGroup = item.Attributes("OptionGroup")

                    If renderedOptionGroups.Contains(currentOptionGroup) Then
                        RenderListItem(item, writer)
                    Else
                        If renderedOptionGroups.Count > 0 Then
                            RenderOptionGroupEndTag(writer)
                        End If

                        RenderOptionGroupBeginTag(currentOptionGroup, writer)
                        renderedOptionGroups.Add(currentOptionGroup)

                        RenderListItem(item, writer)
                    End If
                End If
            Next

            If renderedOptionGroups.Count > 0 Then
                RenderOptionGroupEndTag(writer)
            End If
    End Sub

    Private Sub RenderOptionGroupBeginTag(name As String, writer As HtmlTextWriter)
        writer.WriteBeginTag("optgroup")
        writer.WriteAttribute("label", name)
        writer.Write(HtmlTextWriter.TagRightChar)
        writer.WriteLine()
    End Sub

    Private Sub RenderOptionGroupEndTag(writer As HtmlTextWriter)
        writer.WriteEndTag("optgroup")
        writer.WriteLine()
    End Sub

    Private Sub RenderListItem(item As ListItem, writer As HtmlTextWriter)
        writer.WriteBeginTag("option")
        writer.WriteAttribute("value", item.Value, True)

        If item.Selected Then
            writer.WriteAttribute("selected", "selected", False)
        End If

        For Each key As String In item.Attributes.Keys
            writer.WriteAttribute(key, item.Attributes(key))
        Next

        writer.Write(HtmlTextWriter.TagRightChar)
        HttpUtility.HtmlEncode(item.Text, writer)
        writer.WriteEndTag("option")
        writer.WriteLine()
    End Sub
End Class
```

then in the 'App_Browsers" folder create file "BrowserFile.browser" and add this code:
```
<browsers>
  <browser refID="Default">
    <controlAdapters>
      <adapter controlType="System.Web.UI.WebControls.DropDownList" adapterType="DropDownListAdapter" />
    </controlAdapters>
  </browser>
</browsers>
```

Your program should then understan to use this 'adapter'

To use the dropdown from code you add items like this:
```
Dim li As New ListItem("TEXT", "VALUE")
li.Attributes.Add("OptionGroup", "OPTION GROUP NAME")

DROPDOWNLIST.Items.Add(li)

Dim newLi As New ListItem("TEXT", "VALUE")
newLi.Attributes.Add("OptionGroup", "OPTION GROUP NAME")

newLi.Selected = True

DROPDOWNLIST.Items.Add(newLi)
```

As far as i know asp dropdownlist automatically renders as <select>, so I don't think that all this is necessary, but I know this code does work. 

Good luck

**Comment by seanbotha123** [comment]
Just to confirm a test: @omerzia 

I copied your exact code to my asp project and removed all the code I posted above from my project.

The rendered html looks like so:
![select](https://cloud.githubusercontent.com/assets/5496698/5916295/5a302c7c-a618-11e4-9551-19d02c8928b9.png)

Which is 100% correct.

I then call the bootstrap select function like so:

```
$('select').selectpicker();
```

And this works perfectly.But your way of calling it should also work. maybe check for js conflicts.
